src/quicklist.c

```diff
@@ -718,6 +718,16 @@ void quicklistDelEntry(quicklistIter *iter, quicklistEntry *entry) {
             iter->current = prev;
             iter->offset = -1;
         }
+
+        /* Also updates param 'entry' to the right entry of the deleted one. */
+        entry->node = next;
+        if (entry->node) {
+            quicklistDecompressNodeForUse(entry->node);
```

Comment:
Just want to confirm that it's expected to exit this function with a decompressed node..